test/build/transform-sourcemap/custom-transform-plugin.js

```diff
@@ -0,0 +1,16 @@
+const MagicString = require('magic-string');
+
+module.exports = function () {
+  return {
+    transform: async ({id, fileExt, contents}) => {
+      const ms = new MagicString(contents);
+      ms.appendLeft(contents.indexOf('console.log'), `console.log('transformed');\n`);
+      const map = ms.generateMap({source: id, hires: false, includeContent: true});
+      return {
+        result: ms.toString(),
```

Comment:
nit: Could you change `result` to `contents`?

I know it’s confusing from the current code, but we’ve been trying to move away from `{ result, map }` in favor of `{ contents, map }`. The only reason we have “result“ in our codebase is because that was our first plugin API that we don’t want to break compat for during 2.x.

Anyway, `contents` would better match our APIs going forward (we also have “code” in places but let’s not get into that 😄)